### PR TITLE
Added cusparse capability for bml_transpose_ellpack.

### PR DESCRIPTION
**Pull request by oseikuffuor1**
This PR Adds new functionality for ellpack's transpose using openMP OFFLOAD + cusparse

### CODE REVIEW

**Review by suemni — APPROVED** [review]

File: src/C-interface/ellpack/bml_transpose_ellpack_typed.c [hunk 1]
```diff
@@ -204,5 +217,118 @@ void TYPED_FUNC(
...
+                          omp_get_default_device());
+        omp_target_memcpy(csrValA, cscVal, nnzA * sizeof(REAL_T), 0, 0,
+                          omp_get_default_device(), omp_get_default_device());
+
+        // deallocate storage buffer
+        omp_target_free(dBuffer, omp_get_default_device());
+    }
+
+/*
+// DEBUG:
```
Comment (nicolasbock) [inline src/C-interface/ellpack/bml_transpose_ellpack_typed.c:311]: Please remove commented out code.

**Review by nicolasbock — CHANGES_REQUESTED** [review]

File: src/C-interface/ellpack/bml_transpose_ellpack_typed.c (hunk 1 above)
Comment (oseikuffuor1) [reply src/C-interface/ellpack/bml_transpose_ellpack_typed.c]: @nicolasbock comment has been removed. BTW there are other commented out code that was not introduced in this PR. I left those intact. Thanks.

**Review by nicolasbock — APPROVED** [review]
Thanks!

**Review by jeanlucf22 — APPROVED** [review]

**Comment by jeanlucf22** [comment]
Ready to merge IMO

**Pull request merged**